src/rulesets/Base/array.jl

```diff
@@ -1,3 +1,32 @@
+#####
+##### constructors
+#####
+
+ChainRules.@non_differentiable (::Type{T} where {T<:Array})(::UndefInitializer, args...)
+
+function rrule(::Type{T}, x::AbstractArray) where {T<:Array}
+    project_x = ProjectTo(x)
+    Array_pullback(ȳ) = (NoTangent(), project_x(ȳ))
+    return T(x), Array_pullback
+end
+
+#####
+##### `vect`
+#####
+
+@non_differentiable Base.vect()
+
+# Don't worry about projection here. The data passes straight through, so if a cotangent has
+# the wrong type for some reason, it must be the fault of another rule somewhere.
+function rrule(::typeof(Base.vect), X::T...) where {T}
+    l = length(X)
```

Comment:
Apparent I was wrong about moving this out

https://github.com/JuliaDiff/ChainRules.jl/issues/492#issuecomment-890586186

🤯

Reply:
Haha yup, I noticed that too. I love working with lots of people.